packages/cart/src/services/line-item.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  CartLineItemDTO,
+  Context,
+  DAL,
+  FilterableLineItemProps,
+  FindConfig,
+} from "@medusajs/types"
+import {
+  InjectManager,
+  InjectTransactionManager,
+  MedusaContext,
+  MedusaError,
+  ModulesSdkUtils,
+  isString,
+  retrieveEntity,
+} from "@medusajs/utils"
+import { Cart, LineItem } from "@models"
+import { LineItemRepository } from "@repositories"
+import { CreateLineItemDTO, UpdateLineItemDTO } from "../types"
+import CartService from "./cart"
+
+type InjectedDependencies = {
+  lineItemRepository: DAL.RepositoryService
+  cartService: CartService<any>
+}
+
+export default class LineItemService<
+  TEntity extends LineItem = LineItem,
+  TCart extends Cart = Cart
+> {
+  protected readonly lineItemRepository_: DAL.RepositoryService<LineItem>
+  protected readonly cartService_: CartService<TCart>
+
+  constructor({ lineItemRepository, cartService }: InjectedDependencies) {
+    this.lineItemRepository_ = lineItemRepository
+    this.cartService_ = cartService
+  }
+
+  @InjectManager("lineItemRepository_")
+  async retrieve(
+    id: string,
+    config: FindConfig<CartLineItemDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<TEntity> {
+    return (await retrieveEntity<LineItem, CartLineItemDTO>({
+      id: id,
+      entityName: LineItem.name,
+      repository: this.lineItemRepository_,
+      config,
+      sharedContext,
+    })) as TEntity
+  }
+
+  @InjectManager("lineItemRepository_")
+  async list(
+    filters: FilterableLineItemProps = {},
+    config: FindConfig<CartLineItemDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<TEntity[]> {
+    const queryOptions = ModulesSdkUtils.buildQuery<LineItem>(filters, config)
+
+    return (await this.lineItemRepository_.find(
+      queryOptions,
+      sharedContext
+    )) as TEntity[]
+  }
+
+  @InjectManager("lineItemRepository_")
+  async listAndCount(
+    filters: FilterableLineItemProps = {},
+    config: FindConfig<CartLineItemDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<[TEntity[], number]> {
+    const queryOptions = ModulesSdkUtils.buildQuery<LineItem>(filters, config)
+
+    return (await this.lineItemRepository_.findAndCount(
+      queryOptions,
+      sharedContext
+    )) as [TEntity[], number]
+  }
+
+  @InjectTransactionManager("lineItemRepository_")
+  async create(
+    cartOrId: Cart | string,
```

Comment:
comment: Considering removing cart id argument + cart retrieval in this service method (and update). Instead, let that validation happen at either the module service level or at workflow level

Reply:
Think it would make sense to have that part in the CartService.